gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/util/NumberHelper.java

```diff
@@ -361,8 +386,26 @@ public static Class<? extends Number> getHighestCommonNumberClass(final boolean
      */
     public static Number add(final Number a, final Number b) {
         if (null == a || null == b) return a;
-        final Class<? extends Number> clazz = getHighestCommonNumberClass(a, b);
-        return getHelper(clazz).add.apply(a, b);
+        NumberInfo numberInfo = getHighestCommonNumberInfo(false, a, b);
+        Number result = 0;
+        while (true) {
+            try {
+                final Class<? extends Number> clazz = determineNumberClass(numberInfo.bits, numberInfo.fp);
+                result = getHelper(clazz).add.apply(a, b);
+                if (Double.isInfinite(result.doubleValue()))
+                {
+                    throw new ArithmeticException("Floating point overflow detected");
```

Comment:
I don't think we want an exception in this case, infinity is an acceptable sum result for floating point numbers.